akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -415,6 +416,23 @@ class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Graph
   def dropWhile(p: function.Predicate[Out]): javadsl.Flow[In, Out, Mat] = new Flow(delegate.dropWhile(p.test))
 
   /**
+   * Recover allows to send last element on failure and gracefully complete the stream
+   * Using recover does not prevent from losing messages that arrived before onError
+   * as they pass through this processing step.
```

Comment:
What do you mean by "as they pass through"? I think this should be reformulated to something like this:

"Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements. This stage can recover the failure signal, but not the skipped elements, which will be dropped."